components/email_aliases/email_aliases_service_unittest.cc

```diff
@@ -345,3 +346,364 @@ TEST_F(EmailAliasesServiceTimingTest, VerifyResult_StopsAfterMaxDuration) {
+  }
+
+  base::expected<std::monostate, std::string> CallUpdateAliasWith(
+      const std::string& alias_email,
+      const std::optional<std::string>& put_body,
+      const std::optional<std::string>& refresh_body = std::nullopt,
+      bool wait_for_update = true) {
+    AddManageResponseFor(put_body);
+    AddRefreshResponseFor(refresh_body);
+    auto result_out = InvokeAndWait<std::monostate>([&](auto cb) {
+      service_->UpdateAlias(alias_email, /*note=*/std::string("note"),
+                            std::move(cb));
+    });
+    if (wait_for_update) {
+      EXPECT_TRUE(observer_.WaitForAliasUpdateCount(1));
+    }
+    return result_out;
+  }
+
+  base::expected<std::monostate, std::string> CallDeleteAliasWith(
+      const std::string& alias_email,
+      const std::optional<std::string>& delete_body) {
+    AddManageResponseFor(delete_body);
+    AddRefreshResponseFor(/*refresh_body=*/std::nullopt);
+    auto result_out = InvokeAndWait<std::monostate>(
+        [&](auto cb) { service_->DeleteAlias(alias_email, std::move(cb)); });
+    EXPECT_TRUE(observer_.WaitForAliasUpdateCount(1));
+    return result_out;
+  }
+
+  base::test::ScopedFeatureList feature_list_;
+  base::test::TaskEnvironment task_environment_;
+  network::TestURLLoaderFactory url_loader_factory_;
+  std::unique_ptr<EmailAliasesService> service_;
+  AliasObserver observer_;
+};
+
+namespace {
+
+// Expected alias value used by Generate tests (success path).
+constexpr char kExpectedGeneratedAlias[] = "mock-1234@bravealias.com";
+
+// Build the success response body for Generate tests using the expected alias.
+std::string MakeGenerateAliasSuccessBody() {
+  return std::string("{\"message\":\"created\",\"alias\":\"") +
+         kExpectedGeneratedAlias + "\"}";
+}
+
+struct EmailAliasesApiCase {
+  const char* name;
+  std::optional<std::string> body;
+  // If set, the test expects failure. Holds either a resource id (int) whose
+  // localized string must be contained in the error, or a raw expected string
+  // that must match exactly.
+  std::optional<std::variant<int, std::string>> expected_error;
+};
+
+// Param-to-name converter used by all parameterized suites.
+std::string ParamName(const testing::TestParamInfo<EmailAliasesApiCase>& info) {
+  return info.param.name;
+}
+
+// Shared expectation helper that only validates success/failure state and
+// error substring when applicable. Value checks (on success) are performed
+// by callers when needed.
+template <typename T>
+void ExpectResultStateMatches(const base::expected<T, std::string>& result,
+                              const EmailAliasesApiCase& p) {
+  if (p.expected_error) {
+    ASSERT_FALSE(result.has_value());
+    std::string error_string;
+    if (std::holds_alternative<int>(*p.expected_error)) {
+      const int id = std::get<int>(*p.expected_error);
+      error_string = l10n_util::GetStringUTF8(id);
+    } else {
+      error_string = std::get<std::string>(*p.expected_error);
+    }
```

Comment:
The idiomatic approach is to use `std::visit<>()` on the variant rather than unwrapping it by hand, for example:
```cpp
const std::string error_string =
    std::visit(absl::Overload{
                    [](int id) { return l10n_util::GetStringUTF8(id); },
                    [](std::string string) { return string; },
                },
                *p.expected_error);
```

Reply:
done